File: ApiUrnaEletronicaFront/src/Models/Candidate.ts
import ICandidate from './Interfaces/ICandidate';

export default class Candidate implements ICandidate {
	constructor(
		public fullName: string = '',
		public viceCandidateName: string = '',
		public party: string = '',
		public electoralNumber: number = 0,
		public voteCount: number = 0,
		public id: number = 0
	) {}
}

File: ApiUrnaEletronicaBack/ApiUrnaEletronica/Models/Candidate.cs
using System.ComponentModel.DataAnnotations;

namespace ApiUrnaEletronica.Models
{
    public class Candidate
    {
        [Required]
        [Key]
        public int Id { get; set; }

        [Required]
        public string? FullName { get; set; }

        [Required]
        public string? ViceCandidateName { get; set; }

        [Required]
        public int? ElectoralNumber { get; set; }

        [Required]
        public string? Party { get; set; }

        [Required]
        public DateTime CreatedAt { get; set; }

        public List<Vote>? Votes { get; set; }
    }
}
